src/Api/Parser/AbstractRestParser.php

```diff
@@ -89,18 +89,30 @@ private function extractHeader(
         &$result
     ) {
         $value = $response->getHeaderLine($shape['locationName'] ?: $name);
-        $type = $shape->getType();
 
-        if ($type === 'blob') {
-            $value = base64_decode($value);
-        } elseif ($type === 'timestamp') {
-            try {
-                $value = new DateTimeResult($value);
-            } catch (\Exception $e) {
-                // If the value cannot be parsed, then do not add it to the
-                // output structure.
-                return;
-            }
+        switch ($shape->getType()) {
+            case 'float':
+            case 'double':
+                $value = doubleval($value);
```

Comment:
Using `(double) $value` is faster and is identical in terms of the casting behaviour.